script/ide/System/Database/TableDatabase.lua

```diff
@@ -244,10 +244,60 @@ function TableDatabase:ToData()
 	-- should always return nil;
 end
 
+NPL.load("(gl)script/ide/System/Database/StorageProvider.lua");
+local StorageProvider = commonlib.gettable("System.Database.StorageProvider");
+local config_filename = "/tabledb.config.xml";
 -- automatically called from IO thread
 function TableDatabase:open(rootFolder)
 	self.rootFolder = rootFolder;
+
 	ParaIO.CreateDirectory(rootFolder);
+
+	local config = ParaIO.open(rootFolder..config_filename, "r");
+	if config:IsValid() then
+		local str = config:GetText(0, -1);
+		local xml = ParaXML.LuaXML_ParseString(str);
+
+		NPL.load("(gl)script/ide/System/Compiler/lib/util.lua");
+		local util = commonlib.gettable("System.Compiler.lib.util")
+
+		if xml[1] and xml[1].name == "tabledb" then
+			for i,item in ipairs(xml[1]) do
+				-- providers should always comes first
+				if item.name == "providers" then
+					for i,provider in ipairs(item) do
+						if provider.name == "provider" then
+
+							local t = commonlib.split(provider[1], ",")
+							local init_args = {}
+							if provider.attr.name == "raft" then
+								init_args = {
+									baseDir = t[1],
+									localAddress = {
+										host = t[2],
+										port = t[3],
+										id = t[4],
+									},
+								}
+							end
+
+							NPL.load(provider.attr.file);
+							local storage = commonlib.gettable(provider.attr.type);
+							StorageProvider:RegisterStorageClass(provider.attr.name, storage, init_args)
+
+						end
+					end
+				end
+				if item.name == "tables" then
+					for i,table in ipairs(item) do
+						if table.name == "table" then
+							self:collection(table.attr.name, table.attr.provider);
```

Comment:
BTW, I do not think it is necessary to pass provider here as the second parameter. 
Maybe you can just: collection.parent:FindProvider() on demand. 

And you are also missing `connect` logic when table is opened. 

And I think connect should be async.